geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterQueryServiceCommand.java

```diff
@@ -0,0 +1,209 @@
+      "A comma separated list of all parameter values for the specified method authorizer. Requires '--"
+          + METHOD_AUTHORIZER_NAME + "' option to be used";
+  static final String SINGLE_AUTHORIZER_PARAMETER = "authorizer-parameter";
+  static final String SINGLE_AUTHORIZER_PARAMETER_HELP =
+      "A single parameter to be used by the specified method authorizer. Requires '--"
+          + METHOD_AUTHORIZER_NAME + "' option to be used";
+  static final String DEPRECATED_PROPERTY_ERROR = "Deprecated System Property: \"" + GEMFIRE_PREFIX
+      + "QueryService.allowUntrustedMethodInvocation\" is set to TRUE. In order to use a MethodInvocationAuthorizer, this property must be FALSE or undefined.";
+  static final String PARAMETERS_WITHOUT_AUTHORIZER_MESSAGE =
+      "The '--" + AUTHORIZER_PARAMETERS + "' and '--" + SINGLE_AUTHORIZER_PARAMETER
+          + "' options require '--"
+          + METHOD_AUTHORIZER_NAME + "' to be specified";
+  static final String SINGLE_PARAM_AND_PARAMETERS_SPECIFIED_MESSAGE =
+      "Only one of '--" + AUTHORIZER_PARAMETERS + "' and '--" + SINGLE_AUTHORIZER_PARAMETER
+          + "' options can be specified.";
+  static final String NO_ARGUMENTS_MESSAGE =
+      "No arguments were provided. No changes have been applied.";
+  static final String NO_MEMBERS_FOUND_MESSAGE = "No members found.";
+  static final String SECURITY_NOT_ENABLED_MESSAGE =
+      "Integrated security is not enabled for this distributed system. Updating the method authorizer requires integrated security to be enabled.";
+  public static final String PARTIAL_FAILURE_MESSAGE =
+      "In the event of a partial failure of this command, re-running the command or restarting failing members should restore consistency.";
+
+  @CliCommand(value = COMMAND_NAME, help = COMMAND_HELP)
+  @CliMetaData(
+      interceptor = "org.apache.geode.management.internal.cli.commands.AlterQueryServiceCommand$Interceptor")
+  @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
+      operation = ResourcePermission.Operation.MANAGE)
+  public ResultModel execute(
+      @CliOption(key = METHOD_AUTHORIZER_NAME,
+          help = METHOD_AUTHORIZER_NAME_HELP) String methodAuthorizerName,
+      @CliOption(key = AUTHORIZER_PARAMETERS,
+          help = AUTHORIZER_PARAMETERS_HELP) String[] authorizerParameters,
+      @CliOption(key = SINGLE_AUTHORIZER_PARAMETER,
+          help = SINGLE_AUTHORIZER_PARAMETER_HELP) String singleAuthorizerParameter) {
+    ResultModel result;
+
+    QueryConfigService queryConfigService = getQueryConfigService();
+    Set<String> parametersSet = null;
+    if (methodAuthorizerName != null) {
+      if (!isSecurityEnabled()) {
+        return ResultModel.createError(SECURITY_NOT_ENABLED_MESSAGE);
+      }
+      if (Boolean.parseBoolean(
+          System.getProperty(GEMFIRE_PREFIX + "QueryService.allowUntrustedMethodInvocation"))) {
+        return ResultModel.createError(DEPRECATED_PROPERTY_ERROR);
+      }
```

Comment:
According to my understanding this condition will always be `false`: the property `QueryService.allowUntrustedMethodInvocation` is set on servers only, not on the JVM on which the `gfsh` command is executed, so you can safely remove the whole code block.

Reply:
You're right, this check should be moved to the `AlterQueryServiceFunction` class.